google/cloud/aiplatform/models.py

```diff
@@ -215,10 +244,522 @@ def upload(
             model_name=fields.id, project=fields.project, location=fields.location
         )
 
-    # TODO(b/169782716) add support for deployment when Endpoint class complete
-    def deploy(self):
-        raise NotImplementedError("Deployment not implemented.")
+    def deploy(
+        self,
+        endpoint: Optional["Endpoint"] = None,
+        deployed_model_display_name: Optional[str] = None,
+        traffic_percentage: Optional[int] = 100,
+        machine_type: Optional[str] = None,
+        min_replica_count: Optional[int] = 0,
+        max_replica_count: Optional[int] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = (),
+        timeout: Optional[int] = None,
+    ) -> "Endpoint":
+        """
+        Deploys model to endpoint. Endpoint will be created if unspecified.
+
+        Args:
+            endpoint ("Endpoint"):
```

Comment:
May be obvious to you all, but what happens in the case of a naming conflict?

Reply:
naming conflicts are unlikely if not impossible. they're long computer generated strings.